cubes/new_cube.py

```diff
@@ -0,0 +1,254 @@
+""" JXP VERSION """
+
+'''
+Questions for Clancy:
+
+1. lC vs. C
+2. Order of update_grid()
+3. Survey is unique to a grid, right?
+4. Order of grid parameters
+5. Why 90s per iteration?
+6. Is it ok to get a NAN??   And why is it ignored??
```

Comment:
This is a bug. It is not meant to be ignored, it's because a NAN is generated later anyway. It is OK to get a NAN, since this means "you have 0 chance of detecting this FRB with these parameters". if we wanted to be really smooth about this, we would search for the closest non-zero point in zdm space to that FRB, and assign a large but non-infinite penalty proportional to that distance, to ensure that any minimiser sees a slope to minimise against.